src/auth.js

```diff
@@ -34,7 +35,14 @@ async function retrieveToken(method, client) {
             const githubAudience = core.getInput('jwtGithubAudience', { required: false });
 
             if (!privateKey) {
-                jwt = await core.getIDToken(githubAudience)
+                try {
+                    core.debug("Fetching ID token from GitHub OIDC provider")
+                    jwt = await core.getIDToken(githubAudience)
+                } catch (err) {
+                    const new_error = new Error(`Error fetching ID token from GitHub OIDC provider., message: ${err.message}`);
+                    new_error.stack += err.stack;
+                    throw new_error;
```

Comment:
I accidentally deleted my comment from above because I meant to put it here. It's probably in your email. But I was advocating for just logging the error and rethrowing instead of stacking errors.

Reply:
I think it makes more sense to stack the errors since we're catching all these errors and processing them further up in the stack.

Reply:
Also, we'll get a full stack trace with this method.